en/services/mission.md

```diff
@@ -235,16 +239,89 @@ In more detail, the sequence of operations is:
 1. If no `MISSION_ACK` is received the operation will eventually timeout and may be retried (see [above](#timeout)).
 
 
-### Upload Partial Mission {#upload_partial}
+### Partial Mission Upload {#upload_partial}
 
-TBD
+The diagram below shows the communication sequence to upload a partial mission to a drone (assuming all operations succeed).
 
+> **Note** The message sequence is the same as a [full mission upload](#uploading_mission) except that it is triggered with a [MISSION_WRITE_PARTIAL_LIST](../messages/common.md#MISSION_WRITE_PARTIAL_LIST) instead of a [MISSION_COUNT](../messages/common.md#MISSION_COUNT).
+  An update is expected to be robust (succeed completely or fail and revert to original version).
 
-### Download Partial Mission {#download_partial}
+<span></span>
+> **Note** Unlike for full mission update, partial update is not required to be robust; individual commands can fail.
 
-TBD
+{% mermaid %}
+sequenceDiagram;
+    participant GCS
+    participant Drone
+    GCS->>Drone: MISSION_WRITE_PARTIAL_LIST
+    GCS->>GCS: Start timeout
+    Drone->>GCS: MISSION_REQUEST_INT (n)
+    GCS->>GCS: Start timeout
+    GCS-->>Drone: MISSION_ITEM_INT (n)
+    Drone->>GCS: MISSION_REQUEST_INT (n+1)
+    GCS->>GCS: Start timeout
+    GCS-->>Drone: MISSION_ITEM_INT (n+1)
+    Drone->>GCS: MISSION_ACK
+{% endmermaid %}
+
+In more detail, the sequence of operations is:
+1. GCS (client) sends [MISSION_WRITE_PARTIAL_LIST](../messages/common.md#MISSION_WRITE_PARTIAL_LIST) specifying the start and end index of items to overwrite.
+   - A [timeout](#timeout) must be started for the GCS to wait on the response from Drone (`MISSION_REQUEST_INT`).
+1. Drone (server) receives the message, and prepares to upload mission items.
+1. Drone responds with [MISSION_REQUEST_INT](../messages/common.md#MISSION_REQUEST_INT) requesting the first mission item (`seq==n`).
+1. GCS receives `MISSION_REQUEST_INT` and responds with the requested mission item in a [MISSION_ITEM_INT](../messages/common.md#MISSION_ITEM_INT) message.
+1. Drone receives [MISSION_ITEM_INT](../messages/common.md#MISSION_ITEM_INT) and writes it into the existing mission.
+1. Drone and GCS repeat the `MISSION_REQUEST_INT`/`MISSION_ITEM_INT` cycle, iterating `seq` until all items are uploaded.
```

Comment:
Protocol ambiguity (probably exists in the normal form as well):
  - Are the items required to arrive in sequence (IE `n` must be accepted before `n + 1` can be accepted (this is the ArduPilot impl, although not necessarily correct))
  - If an item is lost in transport who is responsible for rerequesting/sending it. Does the GCS spam until it gets the next request, or does the vehicle keep requesting until it gets the item/gives up? (ArduPilot does the latter)
  - If mission updates are meant to be as atomic/transactional as mentioned above when is the transaction discarded/how do you start a new one cleanly?

Reply:
Thank you.

Yes, the items must arrive in sequence. If an item is received out of sequence the drone will re-request the item it expects. If there is a timeout it will request the missing item. It is up to the vehicle to keep on requesting. The alternative would be for the drone to request all missing items at the end, but since ArduPilot and PX4 both do it this way it makes sense to document this (which I will do)

> If mission updates are meant to be as atomic/transactional as mentioned above when is the transaction discarded/how do you start a new one cleanly?

Good question. I shall evade it - I have made a change that it is non-transactional. I expect a full upload to be sane, but the point of partial upload is to allow simple partial upload of a few points. In this case I think it is reasonable that the GCS should be responsible for sanity checking the upload. If someone wants a safe update, they should do a full upload.

Of course others may disagree. This update is partially about making the definition clear.